nc_py_api/ex_app/providers/task_processing.py

```diff
@@ -44,17 +113,22 @@ def __init__(self, session: NcSessionApp):
         self._session = session
 
     def register(
-        self, name: str, display_name: str, task_type: str, custom_task_type: dict[str, typing.Any] | None = None
+        self,
+        name: str,
+        display_name: str,
+        task_type: str,
```

Comment:
why we still need this and `name` ... etc?

Why we can not just pass the new `TaskProcessingProvider` here?